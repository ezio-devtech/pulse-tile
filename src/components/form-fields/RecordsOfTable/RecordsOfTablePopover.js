import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Spinner from '../../ui-elements/Spinner/Spinner';
import RecordsOfTablePopoverDiagnosis from './RecordsOfTablePopoverDiagnosis';
import RecordsOfTablePopoverMedications from './RecordsOfTablePopoverMedications';
import { fetchPatientDiagnosesDetailRequest } from '../../pages/Diagnosis/ducks/fetch-patient-diagnoses-detail.duck';
import { fetchPatientMedicationsDetailRequest } from '../../pages/Medications/ducks/fetch-patient-medications-detail.duck';
import { patientDiagnosesDetailSelector } from '../../pages/Diagnosis/selectors';
import { patientMedicationsDetailSelector } from '../../pages/Medications/selectors';

import RecordsOfTablePlugins from '../../theme/config/recordsOfTable/recordsOfTablePopover';
import { themeDetailsAction, themeDetailsSelector, themeTypesRecordsDetails } from '../../theme/config/recordsOfTable/recordsOfTableDetails';

const coreActions = {
    fetchPatientDiagnosesDetailRequest,
    fetchPatientMedicationsDetailRequest,
};
const actionsArray = Object.assign(coreActions, themeDetailsAction);
const mapDispatchToProps = dispatch => ({ actions: bindActionCreators(actionsArray, dispatch) });

const coreTypesRecordsDetails = {
  diagnosis: {
    title: 'Diagnosis',
    fetchDetail: 'fetchPatientDiagnosesDetailRequest',
    stateName: 'diagnosisDetail',
  },
  medications: {
    title: 'Medications',
    fetchDetail: 'fetchPatientMedicationsDetailRequest',
    stateName: 'medicationDetail',
  },
};
const typesRecordsDetails = Object.assign(coreTypesRecordsDetails, themeTypesRecordsDetails);

@connect(patientDiagnosesDetailSelector, mapDispatchToProps)
@connect(patientMedicationsDetailSelector)
@connect(themeDetailsSelector)

export default class RecordsOfTablePopover extends PureComponent {

  static propTypes = {
    record: PropTypes.object,
  };

  state = {
    typesRecords: typesRecordsDetails,
    sourceId: '',
  };

  componentDidMount() {
    const { actions, match, record: { type, sourceId } } = this.props;
    const { typesRecords } = this.state;
    const userId = _.get('params.userId', match);
    const fetchRequest = typesRecords[type] ? typesRecords[type].fetchDetail : '';
    if (fetchRequest && userId && sourceId) {
      this.setState({ sourceId });
      actions[fetchRequest]({ userId, sourceId })
    }
  }

  render() {
    const { record: { type: typeOfRecord } } = this.props;
    const { typesRecords, sourceId } = this.state;
    let title = '';
    let detail = null;
    if (typesRecords[typeOfRecord]) {
      title = typesRecords[typeOfRecord].title;
      detail = this.props[typesRecords[typeOfRecord].stateName] || null;
    }
    return (
      <div className="record-popover" style={{ display: 'block' }}>
        <div className="record-popover-header">
          <div className="record-popover-title">{title}</div>
        </div>
        <div className="record-popover-body">
          <div className="record-popover-content">
            {!detail || detail.sourceId !== sourceId ? <Spinner /> : null }
            {typeOfRecord === 'diagnosis' ? <RecordsOfTablePopoverDiagnosis detail={detail} /> : null}
            {typeOfRecord === 'medications' ? <RecordsOfTablePopoverMedications detail={detail} /> : null}
            <RecordsOfTablePlugins typeOfRecord={typeOfRecord} detail={detail} />
          </div>
        </div>
      </div>
    )
  }
}
