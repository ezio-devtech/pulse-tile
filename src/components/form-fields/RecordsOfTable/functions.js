import _ from 'lodash/fp';
import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';

export function changeArraysForTable(arr, name, date) {
  return arr.map((el, index) => {
    el.tableName = el[name];
    el.date = getDDMMMYYYY(el[date]);
    return {
      record: el,
      title: el[name],
      value: index,
    }
  });
}

export function setDiagnosisRecords(data) {
  return changeArraysForTable(data, 'problem', 'dateOfOnset');
}

export function setMedicationsRecords(data) {
  return changeArraysForTable(data, 'name', 'dateCreated');
}
