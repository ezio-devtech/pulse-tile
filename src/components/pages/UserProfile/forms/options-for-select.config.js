const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

export { optionsForThemesField, optionsForGenderField }
