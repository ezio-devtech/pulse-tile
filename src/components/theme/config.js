import vaccinations from './plugins/Vaccinations/index';

/**
 * The list of none-core plugins which are included to the current theme
 *
 * @return array
 */
export const themePlugins = [

    vaccinations,

];

/**
 * The list of clients URLs for none-core plugins
 *
 * @return array
 */
export const themeClientUrls = {

    VACCINATIONS: 'vaccinations',

};

/**
 * The list of pages for none-core plugins
 *
 * @return array
 */
export const themePluginsPages = {

    'vaccinations': {
      breadcrumbs: [{
        title: 'Vaccinations',
        state: '/vaccinations',
      }],
    },

};

/**
 *
 */
export const themePatientSummaryConfig = [
    {
        key: 'vaccinations',
        title: 'Vaccinations',
        titleCheckboxes: 'Vaccinations',
        state: 'vaccinations',
        nameCheckboxes: 'vaccinations',
        imgPreview: '/images/patients-summary/vaccinations.jpg',
        isDefaultSelected: true,
    },
];
