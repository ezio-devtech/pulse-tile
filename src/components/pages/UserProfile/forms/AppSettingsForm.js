import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';

@reduxForm({ form: 'appSettingsFormSelector' })
export default class AppSettingsForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="appSettingsForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Application Title"
                      name="applicationTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Application Logo File"
                      name="logoPath"
                      component={FileInput}
                      id="logoPath"
                    />
                    <Field
                      label="Application Themes"
                      name="green"
                      component={Select}
                      options={optionsForThemesField}
                    />
                    <Field
                      label="Browser Window title"
                      name="browserWindowTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
