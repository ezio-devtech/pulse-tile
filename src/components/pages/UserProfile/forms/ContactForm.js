import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from './validation';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Address"
                      name="address"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="City"
                      name="city"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="State"
                      name="state"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Postal Code"
                      name="postalCode"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Country"
                      name="country"
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Phone Number"
                      name="phone"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Email"
                      name="email"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
