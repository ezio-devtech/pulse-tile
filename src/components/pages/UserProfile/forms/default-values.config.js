import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from './values-names.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
