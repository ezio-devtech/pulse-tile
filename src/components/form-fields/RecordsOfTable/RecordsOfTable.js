import React, { PureComponent } from 'react';
import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
import _ from 'lodash/fp';
import { get } from 'lodash';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SelectFormGroup from '../SelectFormGroup';
import RecordsOfTablePopover from './RecordsOfTablePopover';
import Spinner from '../../ui-elements/Spinner/Spinner';
import { valuesNames, valuesLabels, defaultTypesOptions } from './forms.config';
import { fetchPatientMedicationsRequest } from '../../pages/Medications/ducks/fetch-patient-medications.duck';
import { fetchPatientDiagnosesRequest } from '../../pages/Diagnosis/ducks/fetch-patient-diagnoses.duck';
import { patientDiagnosesSelector } from '../../pages/Diagnosis/selectors';
import { patientMedicationsSelector } from '../../pages/Medications/selectors';

import * as recordsFunctions from './functions';
import { themeActions, themeSelector, themeTypesRecords } from '../../theme/config/recordsOfTable/recordsOfTable';
import { themeFunctions } from '../../theme/config/recordsOfTable/recordsOfTableFunctions';

const PREFIX_POPOVER_ID = 'rot-popover-';

const coreActions = {
  fetchPatientDiagnosesRequest,
  fetchPatientMedicationsRequest,
};
const actionsArray = Object.assign(coreActions, themeActions);
const mapDispatchToProps = dispatch => ({ actions: bindActionCreators(actionsArray, dispatch) });

const coreTypesRecords = {
  diagnosis: {
    title: 'Diagnosis',
    fetchList: 'fetchPatientDiagnosesRequest',
    stateName: 'allDiagnoses',
    setMethodName: 'setDiagnosisRecords',
    records: null,
    hasSubItems: false,
  },
  medications: {
    title: 'Medications',
    fetchList: 'fetchPatientMedicationsRequest',
    stateName: 'allMedications',
    setMethodName: 'setMedicationsRecords',
    records: null,
    hasSubItems: false,
  },
};
const typesRecords = Object.assign(coreTypesRecords, themeTypesRecords);

@connect(patientDiagnosesSelector, mapDispatchToProps)
@connect(patientMedicationsSelector)
@connect(themeSelector)

export default class RecordsOfTable extends PureComponent {

  static defaultProps = {
    typesOptions: defaultTypesOptions,
    isNotDragAndDropOfRaws: false,
    isOnlyOneRecord: false
  };

  state = {
    typeRecords: '',
    indexOfSelectedRecord: '',
    indexOfTypeEvents: '',
    waitingDataOf: '',
    isRecordsLoading: false,
    indexOfOpenedPopover: null,
    isDisplayTypeSelect: true,
    isOnlyOneTypeOfRecords: false,
    typesRecords: typesRecords,
  };

  componentWillMount() {
    const { typesOptions } = this.props;
    if (typesOptions.length === 1) {
      valuesLabels.RECORDS = typesOptions[0].title;
      this.setState({ isDisplayTypeSelect: false, isOnlyOneTypeOfRecords: true });
      this.handleGetHeadingsLists({ target: { value: typesOptions[0].value } });
    }
    window.addEventListener('resize', this.handleClosePopover);
    window.addEventListener('orientationchange', this.handleClosePopover);
    document.addEventListener('click', this.handleDocumentClick);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleClosePopover);
    window.removeEventListener('orientationchange', this.handleClosePopover);
    document.removeEventListener('click', this.handleDocumentClick);
  }

  componentWillReceiveProps(nextProps) {
    const { waitingDataOf } = this.state;
    if (waitingDataOf) {
      this.setState({ isRecordsLoading: false })
    }
    this.setAllRecords(nextProps);
  }

  setAllRecords = (props) => {
    const { typesRecords } = this.state;
    let isShouldUpdate = false;
    const newTypesRecords = {
      ...typesRecords,
    };
    for (const key in newTypesRecords) {
      const stateName = newTypesRecords[key].stateName;
      if (!_.isEmpty(props[stateName]) && _.isEmpty(newTypesRecords[key].records)) {
        const methodName = get(newTypesRecords, '[' + key + '].setMethodName', null);
        let functionName = (typeof(get(recordsFunctions, '[' + methodName+ ']', null)) === 'function')
            ? get(recordsFunctions, '[' + methodName+ ']', null)
            : get(themeFunctions, '[' + methodName+ ']', null);
        newTypesRecords[key].records = (functionName) ? functionName(props[stateName]) : null;
        isShouldUpdate = true;
      }
      this.selectRecordOptionForUpdate(newTypesRecords[key].records);
    }
    if (isShouldUpdate) {
      this.setState({ typesRecords: newTypesRecords });
    }
  };

  selectRecordOptionForUpdate = (typeRecords) => {
    const { isOnlyOneTypeOfRecords } = this.state;
    const { isOnlyOneRecord, input: { value } } = this.props;
    const records = value || [];

    if (isOnlyOneTypeOfRecords && isOnlyOneRecord &&
      typeRecords && typeRecords.length && records.length) {

      const sourceId = records[0].sourceId;

      for (let i = 0; i < typeRecords.length; i++) {
        if (typeRecords[i].record.sourceId === sourceId) {
          this.setState({ indexOfSelectedRecord: typeRecords[i].value});
          break;
        }
      }
    }
  };

  // Functionality of Add and Remove Records items
  handleGetHeadingsLists = (ev) => {
    const { actions, match } = this.props;
    const { typesRecords } = this.state;
    const typeRecords = ev.target.value;
    const userId = _.get('params.userId', match);
    const toSetState = { typeRecords, indexOfSelectedRecord: '', indexOfTypeEvents: '' };

    if (userId && !get(typesRecords, '[' + typeRecords + '].records', null)) {
      toSetState.waitingDataOf = get(typesRecords, '[' + typeRecords + '].stateName', null);
      toSetState.isRecordsLoading = true;
      const fetchList = get(typesRecords, '[' + typeRecords + '].fetchList', null);
      if (fetchList) {
        actions[fetchList]({ userId });
      }
    }

    this.setState(toSetState);
  };

  handleGetHeadingsItems = (ev) => {
    const { input: { onChange, value }, isOnlyOneRecord} = this.props;
    const records = value || [];
    const indexOfSelectedRecord = parseInt(ev.target.value);
    const { typeRecords, typesRecords, indexOfTypeEvents } = this.state;

    const newRecords = records.slice();
    let selectedItem;

    const recordHasSubItem = this.isRecordHasSubitems(typesRecords, typeRecords);
    if (recordHasSubItem) {
      selectedItem = typeRecords ? typesRecords[typeRecords].records[indexOfTypeEvents].events[indexOfSelectedRecord] : null;
    } else {
      selectedItem = typeRecords ? typesRecords[typeRecords].records[indexOfSelectedRecord] : null;
    }

    if (selectedItem) {
      const record = {
        name: get(selectedItem, 'record.tableName', null),
        type: typeRecords,
        typeTitle: typesRecords[typeRecords].title,
        date: get(selectedItem, 'record.date', null),
        source: get(selectedItem, 'record.source', null),
        sourceId: get(selectedItem, 'record.sourceId', null),
      };
      if (recordHasSubItem) {
        record.typeTitle = typeRecords ? typesRecords[typeRecords].records[indexOfTypeEvents].title : null;
      }

      if (isOnlyOneRecord) {
        newRecords[0] = record;
      } else {
        newRecords.push(record);
      }

      onChange(newRecords);
      this.setState({ indexOfSelectedRecord });
    }
  };

  handleGetEventType = (ev) => {
    const indexOfTypeEvents = parseInt(ev.target.value);
    this.setState({ indexOfTypeEvents, indexOfSelectedRecord: '' });
  };

  removeRecord = index => (ev) => {
    ev.preventDefault();
    ev.stopPropagation();
    const { input: { onChange, value } } = this.props;
    const newRecords = value.slice();
    newRecords.splice(index, 1);

    onChange(newRecords);
    this.handleTogglePopover(null);
  };

  // Functionality of Drag and Drop
  reorder = (list, startIndex, endIndex) => {
    const result = Array.from(list);
    const [removed] = result.splice(startIndex, 1);
    result.splice(endIndex, 0, removed);

    return result;
  };

  onDragStart = () => {
    this.handleTogglePopover(null);
  };

  onDragEnd = (result) => {
    // dropped outside the list
    if (!result.destination) { return }
    const { input: { onChange, value } } = this.props;
    const newRecords = this.reorder(value, result.source.index, result.destination.index);
    this.handleTogglePopover(null);
    onChange(newRecords);
  };

  getItemStyle = (isDragging, draggableStyle) => ({
    opacity: isDragging ? 0.5 : 1,
    ...draggableStyle,
  });

  // Functionality of Popover
  handleTogglePopover = (index) => {
    this.setState({ indexOfOpenedPopover: this.state.indexOfOpenedPopover !== index ? index : null })
  };

  handleClosePopover = () => {
    this.handleTogglePopover(null);
  };

  handleDocumentClick = (ev) => {
    const target = ev.target;
    const popoverWrapper = target.closest('.record-popover-wrapper');

    if (!popoverWrapper) {
      this.handleTogglePopover(null);
    }
  };

  isRecordHasSubitems = (typesRecords, typeRecords) => {
    return get(typesRecords, typeRecords + '.hasSubItems', false);
  }

  render() {
    const { typesOptions, isSubmit, input: { value }, match, isNotDragAndDropOfRaws } = this.props;
    const { typesRecords, typeRecords, indexOfSelectedRecord, isRecordsLoading, indexOfTypeEvents, indexOfOpenedPopover, isDisplayTypeSelect } = this.state;
    const records = value;
    const recordHasSubItem = this.isRecordHasSubitems(typesRecords, typeRecords);
    return (
      <div>
        {isRecordsLoading ? <Spinner /> : null }
        {isDisplayTypeSelect ?
          <SelectFormGroup
            label={valuesLabels.TYPE}
            name={valuesNames.TYPE}
            id={valuesNames.TYPE}
            options={typesOptions}
            component={SelectFormGroup}
            placeholder="-- Select type --"
            meta={{ error: false, touched: false }}
            input={{ value: typeRecords }}
            onChange={this.handleGetHeadingsLists}
          /> : null}

          {get(typesRecords, '[' + typeRecords + ']', null) ?
            recordHasSubItem ?
              <div>
                <SelectFormGroup
                  label={valuesLabels.RECORDS_TYPE_EVENTS}
                  name={valuesNames.RECORDS_TYPE_EVENTS}
                  id={valuesNames.RECORDS_TYPE_EVENTS}
                  options={get(typesRecords, '[' + typeRecords + '].records', null) || []}
                  component={SelectFormGroup}
                  placeholder={'-- Select Events Type --'}
                  meta={{ error: false, touched: false }}
                  input={{ value: indexOfTypeEvents }}
                  onChange={this.handleGetEventType}
                />
                { indexOfTypeEvents || indexOfTypeEvents === 0 ?
                  <SelectFormGroup
                    label={valuesLabels.RECORDS_EVENTS}
                    name={valuesNames.RECORDS_EVENTS}
                    id={valuesNames.RECORDS_EVENTS}
                    options={get(typesRecords, '[' + typeRecords + '].records[' + indexOfTypeEvents + '].events', null) || []}
                    component={SelectFormGroup}
                    placeholder={`-- Select ${typesRecords[typeRecords].title} --`}
                    meta={{ error: false, touched: false }}
                    input={{ value: indexOfSelectedRecord }}
                    onChange={this.handleGetHeadingsItems}
                  /> : null
                }
              </div>
              :
              <SelectFormGroup
                label={valuesLabels.RECORDS}
                name={valuesNames.RECORDS}
                id={valuesNames.RECORDS}
                options={get(typesRecords, '[' + typeRecords + '].records', null) || []}
                component={SelectFormGroup}
                placeholder={`-- Select ${typesRecords[typeRecords].title} --`}
                meta={{ error: false, touched: false }}
                input={{ value: indexOfSelectedRecord }}
                onChange={this.handleGetHeadingsItems}
              />
            : null
          }

        { records && records.length ?
          <DragDropContext onDragStart={this.onDragStart} onDragEnd={this.onDragEnd}>
            <Droppable droppableId="droppable">
              {provided => (
                <div
                  className="panel-body-inner-table"
                  ref={provided.innerRef}
                >
                  <div className="form-group">
                    <div className="table table-striped table-hover table-bordered rwd-table table-fixedcol table-records-editable">
                      <div className="table__head">
                        <div className="table__row">
                          <div className="table__col">{valuesLabels.RECORDS_NAME}</div>
                          <div className="table__col table__col-type">{valuesLabels.RECORDS_TYPE}</div>
                          <div className="table__col table__col-date">{valuesLabels.RECORDS_DATE}</div>
                          <div className="table__col table__col-source">{valuesLabels.RECORDS_SOURCE}</div>
                          <div className="table__col table__col-control" />
                        </div>
                      </div>
                      <div className="table__body">
                        { records.map((record, index) =>
                          <Draggable
                            key={`record-${index}`}
                            draggableId={`record-${index}`}
                            index={index}
                          >
                            {(provided, snapshot) => (
                              <div className="table__row-holder record-popover-wrapper">
                                <div
                                  className="table__row"
                                  ref={provided.innerRef}
                                  {...provided.draggableProps}
                                  style={this.getItemStyle(
                                    snapshot.isDragging,
                                    provided.draggableProps.style
                                  )}
                                  onClick={ /* istanbul ignore next */ () => { this.handleTogglePopover(index) }}
                                >
                                  <div
                                    className={`table__col ${!isNotDragAndDropOfRaws ? 'dnd-handle-wrapper' : ''}`}
                                    data-th={valuesLabels.RECORDS_NAME}
                                  >
                                    { !isNotDragAndDropOfRaws ?
                                      <div className="dnd-handle" {...provided.dragHandleProps}>
                                        <i className="fa fa-bars" />
                                      </div>
                                      : null
                                    }
                                    <span>{record[valuesNames.RECORDS_NAME]}</span>
                                  </div>
                                  <div className="table__col table__col-type" data-th={valuesLabels.RECORDS_TYPE}><span>{record[valuesNames.RECORDS_TYPE]}</span></div>
                                  <div className="table__col table__col-date" data-th={valuesLabels.RECORDS_DATE}><span>{record[valuesNames.RECORDS_DATE]}</span></div>
                                  <div className="table__col table__col-source" data-th={valuesLabels.RECORDS_SOURCE}><span>{record[valuesNames.RECORDS_SOURCE]}</span></div>
                                  <div className="table__col table__col-control table-records-editable__control" data-th="">
                                    <div
                                      className="btn btn-smaller btn-danger btn-icon-normal"
                                      onClick={this.removeRecord(index)}
                                    ><i className="btn-icon fa fa-times" /></div>
                                  </div>
                                </div>
                                {provided.placeholder}
                                {index === indexOfOpenedPopover ?
                                  <RecordsOfTablePopover
                                    id={`${PREFIX_POPOVER_ID}${index}`}
                                    record={record}
                                    match={match}
                                  /> : null}
                              </div>
                            )}
                          </Draggable>)}
                        {provided.placeholder}
                      </div>
                    </div>
                  </div>
                </div>
              )}
            </Droppable>
          </DragDropContext>
          : <div className={classNames('form-group', { 'has-error': isSubmit })}>
            <div className="form-control-static">{valuesLabels.RECORDS_NOT_EXIST}</div>
            {isSubmit ? <span className="help-block animate-fade">You must select at least one record.</span> : null}
          </div>
        }
      </div>)
  }
}
