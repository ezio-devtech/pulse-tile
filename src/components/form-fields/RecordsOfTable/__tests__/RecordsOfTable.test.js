import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import RecordsOfTable from '../../../form-fields/RecordsOfTable/RecordsOfTable';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

const userId = '9999999000';

const mockStore = configureStore();
const storeResource = {
  patientsDiagnoses: {
    '9999999000': [
      {
        problem: 'Too bad desease ddd',
        dateOfOnset: 1512432000000,
        source: 'ethercis',
        sourceId: '0234fbd6-bfb5-49b0-bf02-9759a22f471f',
        dateOfOnsetConvert: '05-Dec-2017',
      },
    ],
  },
  patientsMedications: {
    '9999999000': [
      {
        name: 'test medication 11.01',
        doseAmount: 'Dose Amount',
        dateCreated: 1515670672000,
        source: 'ethercis',
        sourceId: '1893365f-4961-4af1-847c-8ccf205aa7a1',
        highlighters: [
          {
            name: 'name',
            status: 'warning',
          },
        ],
        dateCreatedConvert: '11-Jan-2018',
      },
    ],
  },
  patientsReferrals: {
    '9999999000': [
      {
        dateOfReferral: 1511283530634,
        referralFrom: '#Tony Shannon1',
        referralTo: 'Ripplefields Optometry service',
        source: 'ethercis',
        sourceId: '94133578-f505-4e76-b4ed-762462508801',
        dateOfReferralConvert: '21-Nov-2017',
      },
    ],
  },
  patientsVitals: {
    '9999999000': [
      {
        author: 'Dr Tony Shannon',
        dateCreated: 1516367377000,
        newsScore: 4,
        respirationRate: '12.0',
        oxygenSupplemental: 'false',
        heartRate: '45.0',
        temperature: '37.0',
        levelOfConsciousness: 'Voice',
        systolicBP: '112.0',
        diastolicBP: '64.0',
        oxygenSaturation: '97.0',
        source: 'ethercis',
        sourceId: '27ee5e25-4c32-46d2-b45a-f74149d72030',
        id: 1,
        highlighters: [
          {
            name: 'newsScore',
            status: 'success',
          },
        ],
        dateCreatedConvert: '19-Jan-2018',
      },
    ],
  },
  patientsEvents: {
    '9999999000': [
      {
        dateCreated: 1494586220000,
        type: 'Discharge',
        name: 'Discharge to care home',
        description: 'Needs nursing and supervisory care',
        dateTime: 1494496220958,
        source: 'ethercis',
        sourceId: '93ac376d-3ff4-4e0b-b080-47eb3fe81750',
        dateTimeConvert: '11-May-2017',
        sideDateInTimeline: 'right',
      },
    ],
  },
  patientsProcedures: {
    '9999999000': [
      {
        name: 'total replacement of hip',
        date: 1436969493829,
        time: 54693829,
        source: 'ethercis',
        sourceId: 'fa7408c3-7d69-4f50-84ac-cbf735a0ab18',
        dateConvert: '15-Jul-2015',
        timeConvert: '17:11',
      },
    ],
  },
}
const store = mockStore(Object.assign({}, storeResource));
const emptyStore = mockStore({
  diagnosesDetail: {
    // '9999999000': {},
  },
  medicationsDetail: {
    // '9999999000': {},
  },
  referralsDetail: {
    // '9999999000': {},
  },
  vitalsDetail: {
    // '9999999000': {},
  },
  eventsDetail: {
    // '9999999000': {},
  },
  proceduresDetail: {
    // '9999999000': {},
  },
});


const DATE_TO_USE = new Date('2017');
const DATE_TO_USE_TIME = DATE_TO_USE.getTime();
global.Date = jest.fn(() => DATE_TO_USE);
Date.now = jest.fn(() => DATE_TO_USE_TIME);

const match = {
  params: {
    userId,
  },
};

describe('Component <RecordsOfTable />', () => {
  it('should renders with props correctly when type referrals', () => {
    const component = shallow(
      <RecordsOfTable
        store={store}
        input={{
          value: [
            {
              [valuesNames.RECORDS_DATE]: '05-Dec-2017',
              [valuesNames.RECORDS_NAME]: 'Too bad desease ddd',
              [valuesNames.RECORDS_SOURCE]: 'ethercis',
              [valuesNames.SOURCE_ID]: '0234fbd6-bfb5-49b0-bf02-9759a22f471f',
              [valuesNames.TYPE]: 'diagnosis',
              [valuesNames.RECORDS_TYPE]: 'Problems / Diagnosis',
            },
          ],
        }}
        match={match}
      />).dive().dive().dive()
      .dive()
      .dive()
      .dive();

    expect(component.find('Spinner')).toHaveLength(0);
    expect(component.find('SelectFormGroup')).toHaveLength(1);
    expect(component.find('DragDropContext')).toHaveLength(1);
    expect(component.find('Connect(Droppable)')).toHaveLength(1);

    // Testing handleGetHeadingsLists method
    expect(component.state().typeRecords).toEqual('');
    component.instance().handleGetHeadingsLists({ target: { value: 'diagnosis' }});
    expect(component.state().typeRecords).toEqual('diagnosis');

    component.instance().handleGetHeadingsLists({ target: { value: 'events' }});
    expect(component.state().typeRecords).toEqual('events');

    expect(component).toMatchSnapshot();
  });
});

