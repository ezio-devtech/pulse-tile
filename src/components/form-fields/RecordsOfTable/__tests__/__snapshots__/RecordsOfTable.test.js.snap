// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <RecordsOfTable /> should renders with props correctly when type referrals 1`] = `
<div>
  <SelectFormGroup
    component={[Function]}
    id="type"
    input={
      Object {
        "value": "",
      }
    }
    label="Type"
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    name="type"
    onChange={[Function]}
    options={
      Array [
        Object {
          "title": "Problems / Diagnosis",
          "value": "diagnosis",
        },
        Object {
          "title": "Medications",
          "value": "medications",
        },
        Object {
          "title": "Referrals",
          "value": "referrals",
        },
        Object {
          "title": "Events",
          "value": "events",
        },
        Object {
          "title": "Vitals",
          "value": "vitals",
        },
        Object {
          "title": "Procedures",
          "value": "procedures",
        },
      ]
    }
    placeholder="-- Select type --"
  />
  <DragDropContext
    onDragEnd={[Function]}
    onDragStart={[Function]}
  >
    <Connect(Droppable)
      droppableId="droppable"
    />
  </DragDropContext>
</div>
`;
